### DISCUSSION TITLE
Add a "light" geometry

### DISCUSSION THREAD

**Discussion by johnwason** [tesseract-robotics/tesseract#487] [Ideas]
It may be helpful to add a "light" geometry type. For planning this would just be a placeholder, but it would be useful for viewers. We can copy the appropriate parameters from Gazebo.

**Comment by johnwason** [comment]
A "sensor" placeholder might also be useful.

**Reply by Levi-Armstrong** [reply]
Yep, I was thinking about coping over the way ignition is handling the sensors.

**Reply by Levi-Armstrong** [reply]
I believe we could create a Sensor class which inherits from Link that contains additional information about the sensor.

**Reply by Levi-Armstrong** [reply]
http://wiki.ros.org/urdf/XML/sensor

**Reply by johnwason** [reply]
It might be simpler to add additional fields to the Link in addition to visual and collision for sensors and lights. That is how SDF handles it. We should probably model SDF since the URDF sensor definitions aren't used much.

**Reply by johnwason** [reply]
We could also add a bool flag or an enum for the link type, so the system can distinguish kinematic links from reference links.

**Reply by Levi-Armstrong** [reply]
Sounds good. Something like what is shown below?

``` xml
<link name="my_link">
   <inertial>
     <origin xyz="0 0 0.5" rpy="0 0 0"/>
     <mass value="1"/>
     <inertia ixx="100"  ixy="0"  ixz="0" iyy="100" iyz="0" izz="100" />
   </inertial>

   <visual>
     <origin xyz="0 0 0" rpy="0 0 0" />
     <geometry>
       <box size="1 1 1" />
     </geometry>
     <material name="Cyan">
       <color rgba="0 1.0 1.0 1.0"/>
     </material>
   </visual>

   <collision>
     <origin xyz="0 0 0" rpy="0 0 0"/>
     <geometry>
       <cylinder radius="1" length="0.5"/>
     </geometry>
   </collision>

   <sensor name="my_camera_sensor" update_rate="20">
     <camera>
       <image width="640" height="480" hfov="1.5708" format="RGB8" near="0.01" far="50.0"/>
     </camera>
   </sensor>
 </link>

```
